Quality Audit Page Complete

RevatureGentry opened the pull request:
Quality audit page now properly tracks all user actions and persists them properly. Began creation of environment.ts file for more maintainable services. Also implemented Batch Select dropdown menu with search capabilities

RevatureGentry committed:
Quality Audit Page Complete

Quality audit page now properly tracks all user actions and persists them properly. Began creation of environment.ts file for more maintainable services

RevatureGentry committed:
Initial Commit with Central Services

Creating services provided in the root module to be used throughout application

Quinn-Donnelly commented on src/app/Assess-Batch/Services/category.service.ts:
```diff
@@ -27,10 +27,14 @@ export class CategoryService {
   }
+
+  getActiveCatgories(): Observable<Category[]> {
+    return this.http.get<Category[]>(environment.api.category.active, httpOptions);
```
Question around this. If the plan is to fully remove the URLs from the codebase then line 34 should also have something similar to this. Is that the direction we are going?

Quinn-Donnelly replied:
Or is this just a change until the high level services are done.

RevatureGentry replied:
Correct, I am normalizing the URL's in the high level services - making the environment file as new methods are added to each respected service

Quinn-Donnelly reviewed (approved):
I think this is all good just need to determine if we want to do the environment based url building or if we want to do the gRPC or strings in services with the base changing.

Quinn-Donnelly reviewed (approved):
Build is failing though. We need to make sure this build can run `ng build --prod --aot`

Quinn-Donnelly reviewed (changes requested):
See above clicked wrong button

RevatureGentry commented:
Also just realized that categories this week needs to be on the Quality Audit page. Will fix both issues in next commit

RevatureGentry committed:
Savepoint before Order 66

Quick commit before removing the old Assess-Batch module

RevatureGentry committed:
Merge branch 'centralized-services' into quality-audit-module

Merged over earlier work in an attempt to centralize services and models

RevatureGentry committed:
Savepoint before Domain Purge

Centralized all the model/dto objects into domain directory. Fixed bug when calculating average grades.

RevatureGentry committed:
Model/DTO Refactor

Centralized all Model/DTO classes to one, single directory. Repeated implementations led to wild bugs. Begin to create centralized services, will pass prod build tomorrow.

RevatureGentry committed:
Full Model/DTO Refactor

We have a dedicated diretory for all model/data transfer objects. This commit matches first quality-audit-module functionality before model refactor. Prod builds now pass. Beginning to clean up URL's through environment files and centralized services.

Quinn-Donnelly reviewed (approved):
Excellent improvements.
